picard/coverart/image.py

```diff
@@ -469,19 +469,24 @@ def source(self):
         else:
             return 'File %s' % (self.sourcefile)
 
-    def __repr__(self):
-        p = []
-        p.append('%r' % self.sourcefile)
+    def _repr(self):
+        yield '%r' % self.sourcefile
         if self.tag is not None:
-            p.append("tag=%r" % self.tag)
+            yield 'tag=%r' % self.tag
         if self.types:
-            p.append("types=%r" % self.types)
+            yield 'types=%r' % self.types
         if self.is_front is not None:
-            p.append("is_front=%r" % self.is_front)
-        p.append('support_types=%r' % self.support_types)
+            yield 'is_front=%r' % self.is_front
+        yield 'support_types=%r' % self.support_types
         if self.comment:
-            p.append("comment=%r" % self.comment)
-        return "%s(%s)" % (self.__class__.__name__, ", ".join(p))
+            yield 'comment=%r' % self.comment
```

Comment:
The parent implementation has a different order for is _front and _support_types and also outputs support_multi_types. I wonder if we should instead call the parent implementation:

```python
def _repr(self):
    yield '%r' % self.sourcefile
    if self.tag is not None:
        yield 'tag=%r' % self.tag
    yield from super()._repr()
```

The only additional field handled by the parent function is `url`, but if that is empty it is skipped.

Reply:
Done (and I did more or less the same for _str()) in [c8a41de](https://github.com/metabrainz/picard/pull/2319/commits/c8a41de4103e631164241fd81f00ee46f14e4022)